File: Backend/ApiApp/Models/Product.cs
namespace ApiApp.Models
{
	public class Product
	{
		public int Id { get; set; }
		public string? Name { get; set; }
		public string? Description { get; set; }
		public decimal Price { get; set; }
		public string? ImageUrl { get; set; }
		public int Stock { get; set; }
		public string? Author { get; set; }
		public string? Publisher { get; set; }
		public int? Rating { get; set; }
		public string? Category { get; set; }
	}
}

File: Frontend/src/common/models/product.ts
export class Product {
	id?: number;
	name?: string;
	description?: string;
	price?: number;
	imageUrl?: string;
	stock?: number;
	auther?: string;
	rating?: number;
	publisher?: string;
	category?: string;
}

export class ProductQueryRequest {
	name?: string;
	category?: string;
	rating?: number;
}